### PR TITLE
enh(dart) Add `base`, `interface`, `sealed`, and `when` keywords

### PR DESCRIPTION
**Pull request by srawlins**
Fixes https://github.com/highlightjs/highlight.js/issues/3725

### Changes

Added three keywords, and corrected a typo in `inferface`

### Checklist
- [ ] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by srawlins** [comment]
Is this a simple enough change to not warrant tests, or should I add some?  In the past when adding keywords I have not added tests 🤷 .

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ Grammars:
 - added 3rd party Motoko grammar to SUPPORTED_LANGUAGES [rvanasa][]
 - added 3rd party Candid grammar to SUPPORTED_LANGUAGES [rvanasa][]
 - fix(haskell) Added support for characters [CrystalSplitter][]
+- enh(dart) Add `base`, `interface`, `sealed`, and `when` keywords [Sam Rawlins][]
 
 [AdamRaichu]: https://github.com/AdamRaichu
 [Ali Ukani]: https://github.com/ali
```
Comment (joshgoebel) [inline CHANGES.md:18]: Please also add yourself here, for the linking...

**Review by joshgoebel — APPROVED** [review]

File: CHANGES.md (hunk 1 above)
Comment (srawlins) [reply CHANGES.md:18]: I would but I'm already on line 1051

Comment (joshgoebel) [reply CHANGES.md:18]: You have to be In each release section, the file is split for releases.

Comment (srawlins) [reply CHANGES.md:18]: Ah I see. Fair enough, done!

**Pull request merged**